uwb_localizer_node:
  ros__parameters:
    # Serial communication parameters
    port: "/dev/ttyUSB1"
    baud: 460800
    timeout: 1.0

    publication_frequency: 10.0 # Hz

    # Frame and coordinate system
    frame_id: "uwb_localizer"
    z_sign: 1

    # Timer frequency (Hz)
    timer_frequency: 100.0

    # Range sensor parameters
    min_range: 0.0
    max_range: 100.0
    field_of_view: 0.1
    radiation_type: 1  # INFRARED

    # Calibration parameters
    calibration: "none"  # linear, quadratic, cubic, none
    calib_params: [1.0, 0.0]  # [a, b] for linear, [a, b, c] for quadratic, [a, b, c, d] for cubic

    # Anchor positions (x, y, z in meters)
    anchors_names: ["anchor.1", "anchor.2", "anchor.3", "anchor.4"]
    anchor.1: [0.0, 0.0, 0.0]
    anchor.2: [3.0, 0.0, 0.0]
    anchor.3: [0.0, 3.0, 0.0]
    anchor.4: [3.0, 3.0, 0.0]
